### PR TITLE
fix: Replace deprecated method from Updatable interface

### LINKED ISSUE #3188: Replace deprecated method from Updatable interface
### Component

None

### Description
Method `Updatable.replace` is used in several places in the project. However, the method is deprecated, and Javadoc advises using the method `update` instead. The reason behind it is explained in [alternatives-to-createOrReplace-and-replace](https://github.com/fabric8io/kubernetes-client/blob/main/doc/FAQ.md#alternatives-to-createOrReplace-and-replace).

### PR DESCRIPTION
**Pull request by abusk**
## Description

Fixes #3188 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java [hunk 1]
```diff
@@ -160,21 +160,21 @@ private void updateImageName(KubernetesClient kubernetes, HasMetadata entity, St
             Deployment resource = (Deployment) entity;
             DeploymentSpec spec = resource.getSpec();
             if (spec != null && updateImageName(entity, spec.getTemplate(), imagePrefix, imageName)) {
-                kubernetes.apps().deployments().inNamespace(namespace).resource(resource).replace();
+                kubernetes.apps().deployments().inNamespace(namespace).resource(resource).update();
```
Comment (manusa) [inline jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java:163]: In order to properly swap the deprecated `replace` call with `update` the resource should at least be unlocked to ensure that no optimistic locking breaks the current behavior:

```java
/* ... */.resource(resource).unlock().update()
```

Comment (abusk) [reply jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java]: made the necessary changes.

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**